subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractSmokeTest.groovy

```diff
@@ -112,7 +112,9 @@ abstract class AbstractSmokeTest extends Specification {
         })
 
         // https://plugins.gradle.org/plugin/org.gretty
-        static gretty = "3.0.6"
+        static gretty = JavaVersion.current() < JavaVersion.VERSION_11 ? "3.0.7" : "4.0.0"
```

Comment:
You could use a list here, instead of setting this magically based on the Java version.